trino/trino.go

```diff
@@ -1043,6 +1115,30 @@ func (qr *driverRows) initColumns(qresp *queryResponse) error {
 	return nil
 }
 
+func (qr *driverRows) updateCallback(id string, stats stmtStats) {
+	if qr.stmt.conn.callback != nil {
+		qrStats := QueryProgressCallbackResponse{
+			ID:         id,
+			QueryStats: stats,
+		}
+		currentTime := time.Now()
+		diff := currentTime.Sub(qr.stmt.conn.callbackPeriod.LastCallbackTime)
+		period := qr.stmt.conn.callbackPeriod.Period
+
+		// Check if query state changes OR if period has passed
+		if qr.stmt.conn.callbackPeriod.LastQueryState != qrStats.QueryStats.State || diff >= period {
```

Comment:
Why do you think `qr.stmt.conn.callbackPeriod.LastQueryState != qrStats.QueryStats.State` is valuable? I would inform even if query is stalling.  This is also a piece of information.

Reply:
This condition triggers an update when there _was_ a state change but not enough time passed since the last update.